File: test/epddl_requirements_typing_test.exs
defmodule EpddleRequirementsTypingTest do

  use ExUnit.Case
  doctest EPDDL

  require Record

  test "Types property implies typing requirement" do
    domainStr = "(define (domain foo) (:types bar))"
    domainDef = EPDDL.parse(domainStr)
    assert(EPDDL.domain(domainDef, :requirements) == ["typing"])
  end

  test "Typed vars imply typing requirement" do
    domainStr = "(define (domain foo) (:predicates (at ?v - vehicle)))"
    domainDef = EPDDL.parse(domainStr)
    assert(EPDDL.domain(domainDef, :requirements) == ["typing"])
  end

  test "Types and Typed vars imply typing requirement" do
    domainStr = "(define (domain foo) (:types vehicle) (:predicates (at ?v - vehicle)))"
    domainDef = EPDDL.parse(domainStr)
    assert(EPDDL.domain(domainDef, :requirements) == ["typing"])
  end

end

File: test/epddl_requirements_typing_test.erl
-module(epddl_requirements_typing_test).

-include("../include/epddl_types.hrl").

-include_lib("eunit/include/eunit.hrl").

types_forces_typing_requirement_test() ->
    DomainStr = "(define (domain foo) (:types bar))",
    Domain = epddl:parse(DomainStr),
    Domain#domain.requirements == ["typing"].

typed_vars_forces_typing_requirement_test() ->
    DomainStr = "(define (domain foo) (:predicates (bar ?v - vehicle)))",
    Domain = epddl:parse(DomainStr),
    Domain#domain.requirements == ["typing"].

types_and_typed_vars_forces_typing_requirement_test() ->
    DomainStr = "(define (domain foo) (:types vehicle) (:predicates (bar ?v - vehicle)))",
    Domain = epddl:parse(DomainStr),
    Domain#domain.requirements == ["typing"].
